### PR TITLE
enchancement/account-page

### PR DESCRIPTION
**Pull request by ria0308**

### CODE REVIEW

File: src/components/common/ui/form/input/index.ts
Comment (Markusplay) [inline src/components/common/ui/form/input/index.ts:1]: wtf

File: src/components/pages/account-page/components/general-tab/GeneralTab.module.scss
Comment (Markusplay) [inline src/components/pages/account-page/components/general-tab/GeneralTab.module.scss:1]: remove comments

File: src/components/pages/account-page/components/general-tab/GeneralTab.styles.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  a: {
+    flexGrow: 1,
+  },
+  '& button': {
+    marginTop: '10px',
+  },
+};
+
+export const addSocialLinksContainer: SxProps<Theme> = {
+  border: '2px solid $background-dark-35',
```
Comment (Markusplay) [inline src/components/pages/account-page/components/general-tab/GeneralTab.styles.ts:73]: use color from palette

File: src/components/pages/account-page/components/general-tab/components/contacts-block/constants/index.ts
Comment (Markusplay) [inline src/components/pages/account-page/components/general-tab/components/contacts-block/constants/index.ts:1]: I would say it is "validation" folder, not "constants"

File: src/components/pages/account-page/components/group-tab/GroupTab.tsx [hunk 2]
```diff
@@ -70,15 +61,16 @@ const GroupTab: FC = () => {
...
-      </div>
+    <Box>
+      <Box>
+        <Typography variant="h4">Список групи {user.group.code}</Typography>
+        <Divider
+          sx={stylesMui.groupDivider}
+          textAlign={DividerTextAlign.LEFT}
+          text={`Студенти`}
+        />
+      </Box>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/group-tab/GroupTab.tsx:73]: Do we really need this `Box` wrap here ?

File: src/components/pages/account-page/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  GROUP = 'group',
+  SELECTIVE = 'selective',
+}
+
+export enum AccountPagesMapper {
+  group = 'Група',
+  security = 'Безпека',
+  general = 'Загальне',
+  selective = 'Мої вибіркові',
+}
```
Comment (Markusplay) [inline src/components/pages/account-page/types/index.ts:13]: Due to enum naming conventions, as you did above, make keys with upper case too.

File: src/components/pages/account-page/components/selective-tab/components/opened-selective/OpenedSelective.tsx
Comment (Markusplay) [inline src/components/pages/account-page/components/selective-tab/components/opened-selective/OpenedSelective.tsx:1]: If you don't see what refactor, better to remove this todo than adding new comment

File: src/components/pages/account-page/components/security-tab/components/change-password-form/ChangePasswordForm.styles.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  [theme.breakpoints.down('desktopSemiMedium')]: {
+    display: 'block',
+  },
+};
+export const changePasswordButton: SxProps<Theme> = {
+  fontSize: '16px',
+  [theme.breakpoints.down('desktopSemiMedium')]: {
+    fontSize: '14px',
+  },
+};
```
Comment (Markusplay) [inline src/components/pages/account-page/components/security-tab/components/change-password-form/ChangePasswordForm.styles.ts:26]: use typography

File: src/components/pages/account-page/components/security-tab/SecurityTab.tsx [hunk 5]
```diff
@@ -23,54 +26,47 @@ const SecurityTab = () => {
...
+            value={user.username}
+          />
+          <Input
+            readOnly
+            onChange={() => {}}
+            label="Пошта"
+            value={user.email}
+          />
+        </Box>
+        <Divider sx={styles.divider}></Divider>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/security-tab/SecurityTab.tsx:59]:
```suggestion
        <Divider sx={styles.divider}/>
      
```

File: src/components/pages/account-page/components/security-tab/SecurityTab.tsx [hunk 6]
```diff
@@ -23,54 +26,47 @@ const SecurityTab = () => {
...
+            text={'Зміна паролю'}
+            textAlign={DividerTextAlign.LEFT}
+          ></Divider>
+          <ChangePasswordForm />
+        </Box>
+        <Divider
+          sx={styles.divider}
+          text={'Юзернейм і пошта'}
+          textAlign={DividerTextAlign.LEFT}
+        ></Divider>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/security-tab/SecurityTab.tsx:43]: make it self close

File: src/components/pages/account-page/components/security-tab/SecurityTab.tsx [hunk 7]
```diff
@@ -23,54 +26,47 @@ const SecurityTab = () => {
...
-        <div className={styles['button']}></div>
-      </div>
-      <div className={styles['button-container']}>
+    <Box>
+      <Box>
+        <Box>
+          <Divider
+            text={'Зміна паролю'}
+            textAlign={DividerTextAlign.LEFT}
+          ></Divider>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/security-tab/SecurityTab.tsx:36]: make it self close

File: src/components/pages/account-page/components/security-tab/SecurityTab.tsx [hunk 8]
```diff
@@ -23,54 +26,47 @@ const SecurityTab = () => {
...
-          value={user.username}
-        />
-        <Input readOnly onChange={() => {}} label="Пошта" value={user.email} />
-      </div>
-      <div className={styles['division']}>
-        <div className={styles['white']}></div>
-        <div className={styles['button']}></div>
-      </div>
-      <div className={styles['button-container']}>
+    <Box>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/security-tab/SecurityTab.tsx:30]: remove extra wrapper

File: src/components/pages/account-page/components/security-tab/SecurityTab.styles.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+export const container: SxProps<Theme> = {
+  width: '100%',
+};
+
+export const divider: SxProps<Theme> = {
+  paddingBottom: '20px',
+};
+
+export const button: SxProps<Theme> = {
+  padding: '12px 44.5px',
```
Comment (Markusplay) [inline src/components/pages/account-page/components/security-tab/SecurityTab.styles.ts:15]: make it 12px 45px :)

File: src/components/pages/account-page/components/group-tab/components/no-group-block/NoGroupBlock.tsx [hunk 10]
```diff
@@ -40,37 +41,39 @@ const NoGroupBlock: FC = () => {
...
-            <h4 className={styles['division-text']}>Або виберіть іншу групу</h4>
-            <div className={styles['white']}></div>
-            <div className={styles['button']}></div>
-          </div>
+          </Box>
+          <Box>
+            <Box sx={muiStyles.divisionWhite}></Box>
+            <Typography variant="h4">Або виберіть іншу групу</Typography>
+            <Box sx={muiStyles.divisionWhite}></Box>
+            <Box></Box>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/group-tab/components/no-group-block/NoGroupBlock.tsx:62]: wtf

File: src/components/pages/account-page/components/group-tab/components/no-group-block/NoGroupBlock.tsx [hunk 11]
```diff
@@ -40,37 +41,39 @@ const NoGroupBlock: FC = () => {
...
-            <div className={styles['white']}></div>
-            <h4 className={styles['division-text']}>Або виберіть іншу групу</h4>
-            <div className={styles['white']}></div>
-            <div className={styles['button']}></div>
-          </div>
+          </Box>
+          <Box>
+            <Box sx={muiStyles.divisionWhite}></Box>
+            <Typography variant="h4">Або виберіть іншу групу</Typography>
+            <Box sx={muiStyles.divisionWhite}></Box>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/group-tab/components/no-group-block/NoGroupBlock.tsx:61]:
```suggestion
            <Box sx={muiStyles.divisionWhite}/>
```

File: src/components/pages/account-page/components/group-tab/components/no-group-block/NoGroupBlock.tsx [hunk 12]
```diff
@@ -40,37 +41,39 @@ const NoGroupBlock: FC = () => {
...
-          </div>
-          <div className={styles['division']}>
-            <div className={styles['white']}></div>
-            <h4 className={styles['division-text']}>Або виберіть іншу групу</h4>
-            <div className={styles['white']}></div>
-            <div className={styles['button']}></div>
-          </div>
+          </Box>
+          <Box>
+            <Box sx={muiStyles.divisionWhite}></Box>
```
Comment (Markusplay) [inline src/components/pages/account-page/components/group-tab/components/no-group-block/NoGroupBlock.tsx:59]:
```suggestion
            <Box sx={muiStyles.divisionWhite}/>
```

**Review by Markusplay — CHANGES_REQUESTED** [review]

File: src/components/pages/account-page/components/general-tab/components/personal-info/PersonalInfo.tsx [hunk 13]
```diff
@@ -53,38 +52,20 @@ const PersonalInfoBlock: FC = () => {
...
+            <Input sx={styles.input} label="Ім'я" name="firstName" />
+            <Input sx={styles.input} label="По батькові" name="middleName" />
+            <Box sx={styles.confirmButton}>
               <Button
                 text="Зберегти зміни"
                 startIcon={<CustomCheck />}
-                size={ButtonSize.SMALL}
+                size={isMobile ? ButtonSize.SMALL : ButtonSize.MEDIUM}
                 type="submit"
+                sx={isMobile ? styles.buttonPadding : {}}
```
Comment (Markusplay) [inline src/components/pages/account-page/components/general-tab/components/personal-info/PersonalInfo.tsx:65]:
```suggestion
                sx={isMobile && styles.buttonPadding}
```

Comment (Yevheniy2020) [reply src/components/pages/account-page/components/general-tab/components/personal-info/PersonalInfo.tsx]: image

**Review by Markusplay — APPROVED** [review]

**Review by guywhodiesfirst — COMMENTED** [review]
very small margin between header and students list
also, the font of header should be bold
StudentList
StudentList2

**Comment by guywhodiesfirst** [comment]
some students profile pics are displayed incorrectly - depending on page scale, they are compressed vertically or horizontally
vfkbq
vfkbq1

**Comment by IK-24KudinaDiana** [comment]
Home -> Security
there should be no validation on the password field, validation of the password field does not work correctly. (see screenshots)
On prod: there may be a Ukrainian-English password, but it cannot be updated to the same Ukrainian-English one
photo_2023-09-26_23-37-12
photo_2023-09-26_23-37-01
photo_2023-09-26_23-36-44
photo_2023-09-26_23-36-39
photo_2023-09-26_23-36-35

**Comment by kovaliovev** [comment]
Request card does not match the [design](https://www.figma.com/file/FHBiKogMc9VzGWojaQxA9c/%F0%9F%94%A52.1?type=design&node-id=3230-24590&mode=design&t=JUQkIyIF9ChewhBv-4).
photo_2023-09-28_20-18-16
Screenshot_1

**Comment by kovaliovev** [comment]
Opportunity to give student with role "Заст. старости" this role again.
image

**Comment by kovaliovev** [comment]
Request denied error does not match the [design](https://www.figma.com/file/FHBiKogMc9VzGWojaQxA9c/%F0%9F%94%A52.1?type=design&node-id=3270-25036&mode=design&t=JUQkIyIF9ChewhBv-4).
image
image

**Comment by kovaliovev** [comment]
Footer does not match the [design](https://www.figma.com/file/FHBiKogMc9VzGWojaQxA9c/%F0%9F%94%A52.1?type=design&node-id=2774-21969&mode=design&t=JUQkIyIF9ChewhBv-4).
image
image

**Comment by kovaliovev** [comment]
Too long lines on Security tab. [design](https://www.figma.com/file/FHBiKogMc9VzGWojaQxA9c/%F0%9F%94%A52.1?type=design&node-id=1584-8060&mode=design&t=oJ4Sx8TyznHyCyqO-4).
Screenshot_1
Screenshot_2

**Comment by kovaliovev** [comment]
Too long input fields on Security tab on mobile.
image

**Comment by IK-24KudinaDiana** [comment]
photo_2023-10-02_10-39-11
The headman cannot become a deputy headman

**Comment by IK-24KudinaDiana** [comment]
figma
real result
The first screenshot is a figma, the second is the site.
Inconsistency in the name and appearance of the buttons

**Comment by IK-24KudinaDiana** [comment]
Знімок екрана 2023-10-02 110018
Inconsistency with figma template: no icon, no username, close button too low.

**Comment by IK-24KudinaDiana** [comment]
photo_2023-10-02_11-28-39
The first letter is small.

**Comment by IK-24KudinaDiana** [comment]
photo_2023-10-02_12-06-47
There is a template test under the password, but there is none on the site.

**Comment by guywhodiesfirst** [comment]
there is still a problem when some user profile pics are stretching depending on a page scale
https://github.com/fictadvisor/fictadvisor-web/assets/144242408/26bb2660-7492-489d-ac0e-3cf3224c7038

**Comment by kovaliovev** [comment]
First text is center-align, when second is left-align.
image

**Comment by kovaliovev** [comment]
No margin between emails and buttons at 1200-1225 px window width.
image

**Comment by kovaliovev** [comment]
Group tab does not match the [design](https://www.figma.com/file/FHBiKogMc9VzGWojaQxA9c/%F0%9F%94%A52.1?type=design&node-id=1922-11604&mode=design&t=TL9EfZ0E7cqtu2aU-4).
image
image

**Comment by kol-oss** [comment]
**Button("Додати посилання") incorrectly changes size on the breakpoint 1440**

**SIZE:** 1440 * 1020;

**Video:**

https://github.com/fictadvisor/fictadvisor-web/assets/62239358/bcb33051-96d0-49d8-8dbf-61747dc6c828

**Actual (deploy):**

image

image

**Expected ([figma](https://www.figma.com/file/FHBiKogMc9VzGWojaQxA9c/%F0%9F%94%A52.1?type=design&node-id=299%3A58&mode=design&t=1SwkEEGLo80oTnBV-1)):**

image

**Pull request merged**